physics/ephemeris_body.hpp

```diff
@@ -371,13 +371,17 @@ bool Ephemeris<Frame>::FlowWithAdaptiveStep(
     not_null<DiscreteTrajectory<Frame>*> const trajectory,
     IntrinsicAcceleration intrinsic_acceleration,
     Instant const& t,
-    AdaptiveStepParameters const& parameters) {
+    AdaptiveStepParameters const& parameters,
+    std::int64_t const max_ephemeris_steps =
+        std::numeric_limits<std::int64_t>::max()) {
   std::vector<not_null<DiscreteTrajectory<Frame>*>> const trajectories =
       {trajectory};
   std::vector<IntrinsicAcceleration> const intrinsic_accelerations =
       {std::move(intrinsic_acceleration)};
-  if (empty() || t > t_max()) {
-    Prolong(t);
+  Instant const t_final =
+      std::min(t_max() + max_ephemeris_steps * parameters_.step(), t);
```

Comment:
Doing arithmetic on `max_ephemeris_steps`, which can be `max()`, is going to be problematic.